verifier/trustpolicy/blob.go

```diff
@@ -0,0 +1,135 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trustpolicy
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/notaryproject/notation-go/dir"
+	"github.com/notaryproject/notation-go/internal/slices"
+)
+
+// BlobDocument represents a trustpolicy.blob.json document
+type BlobDocument struct {
+	// Version of the policy document
+	Version string `json:"version"`
+
+	// BlobTrustPolicies include each policy statement
+	BlobTrustPolicies []BlobTrustPolicy `json:"trustPolicies"`
+}
+
+type BlobTrustPolicy struct {
+	// Name of the policy statement
+	Name string `json:"name"`
+
+	// SignatureVerification setting for this policy statement
+	SignatureVerification SignatureVerification `json:"signatureVerification"`
+
+	// TrustStores this policy statement uses
+	TrustStores []string `json:"trustStores"`
+
+	// TrustedIdentities this policy statement pins
+	TrustedIdentities []string `json:"trustedIdentities"`
+
+	// GlobalPolicy defines if policy statement is global or not
+	GlobalPolicy bool `json:"globalPolicy,omitempty"`
+}
+
+// LoadBlobDocument loads a trust policy document from a local file system
+func LoadBlobDocument() (*BlobDocument, error) {
+	var doc BlobDocument
+	err := getDocument(dir.PathBlobTrustPolicy, &doc)
+	return &doc, err
+}
+
+// Validate validates a policy document according to its version's rule set.
+// if any rule is violated, returns an error
+func (policyDoc *BlobDocument) Validate() error {
+	// sanity check
+	if policyDoc == nil {
+		return errors.New("trust policy document cannot be nil")
```

Comment:
This applies to other error messages in this PR, can we add the word `blob` in the error message? This makes the error message very clear to the user that the blob trust policy is malformed. Something like:
```suggestion
		return errors.New("blob trust policy document cannot be nil")
```